ext/hal/nordic/nrfx_config_nrf51.h

```diff
@@ -0,0 +1,1583 @@
+#ifndef NRFX_CONFIG_NRF51_H__
```

Comment:
Need to add a license header

Reply:
Right. I was considering the "nrfx_config_*.h" files as parts of the "nrfx_config.h" file, which has a proper license header. But I think it was not a good approach. I'll add the headers.